### PR TITLE
Refactor ProtoQueryResponseFactory::createTransactionsPageResponse

### PR DESCRIPTION
**Pull request by Exctues**
Description of the Change
Refactored with boost::optional so that there is only 1 method ProtoQueryResponseFactory::createTransactionsPageResponse

Benefits
No duplicate code

Possible Drawbacks
None

### CODE REVIEW

File: shared_model/interfaces/iroha_internal/query_response_factory.hpp [hunk 1]
```diff
@@ -154,33 +154,19 @@ namespace shared_model {
...
-       * @param query_hash - hash of the query, for which response is created
+       * @param next_tx_hash - hash of the transaction after
        * @return transactions response
        */
       virtual std::unique_ptr<QueryResponse> createTransactionsPageResponse(
           std::vector<std::unique_ptr<shared_model::interface::Transaction>>
               transactions,
           interface::types::TransactionsNumberType all_transactions_size,
-          const crypto::Hash &query_hash) const = 0;
+          const crypto::Hash &query_hash,
```
Comment (lebdron) [inline shared_model/interfaces/iroha_internal/query_response_factory.hpp]: Please make `query_hash` the last parameter to be consistent with other methods.

File: shared_model/backend/protobuf/proto_query_response_factory.hpp [hunk 2]
```diff
@@ -6,6 +6,8 @@
 #ifndef IROHA_PROTO_QUERY_RESPONSE_FACTORY_HPP
 #define IROHA_PROTO_QUERY_RESPONSE_FACTORY_HPP
 
+#include "boost/none.hpp"
```
Comment (lebdron) [inline shared_model/backend/protobuf/proto_query_response_factory.hpp]: No need to reference `none` in the header file.

File: shared_model/interfaces/iroha_internal/query_response_factory.hpp [hunk 3]
```diff
@@ -154,33 +154,19 @@ namespace shared_model {
...
+          boost::optional<const crypto::Hash &> next_tx_hash) const = 0;
```
Comment (lebdron) [inline shared_model/interfaces/iroha_internal/query_response_factory.hpp]: And probably make `next_tx_hash` the second parameter again for consistency with previous implementation.

File: shared_model/interfaces/iroha_internal/query_response_factory.hpp (hunk 1 above)
Comment (Exctues) [reply shared_model/interfaces/iroha_internal/query_response_factory.hpp]: Done.

File: shared_model/interfaces/iroha_internal/query_response_factory.hpp (hunk 3 above)
Comment (Exctues) [reply shared_model/interfaces/iroha_internal/query_response_factory.hpp]: Done.

File: shared_model/backend/protobuf/proto_query_response_factory.hpp (hunk 2 above)
Comment (Exctues) [reply shared_model/backend/protobuf/proto_query_response_factory.hpp]: Even optional is not needed here. Removed both.

File: shared_model/interfaces/iroha_internal/query_response_factory.hpp [hunk 4]
```diff
@@ -154,31 +154,17 @@ namespace shared_model {
       /**
        * Create response for transactions pagination query
        * @param transactions - list of transactions in this page
-       * @param next_tx_hash - hash of the transaction after
        * the last in the page
        * @param all_transactions_size - total number of transactions
        * for this query
        * @param query_hash - hash of the query, for which response is created
+       * @param next_tx_hash - hash of the transaction after
```
Comment (MBoldyrev) [inline shared_model/interfaces/iroha_internal/query_response_factory.hpp]: wrong *param*s order?

Comment (Exctues) [reply shared_model/interfaces/iroha_internal/query_response_factory.hpp]: Fixed.

**Comment by lebdron** [comment]
multisig_tx_pipeline_test is known to be nondeterministic in master branch.

**Pull request merged**